src/Native/Unix/System.Native/pal_io.c

```diff
@@ -163,6 +163,49 @@ static void ConvertFileStatus(const struct stat_* src, struct FileStatus* dst)
 #endif
 }
 
+static void ConvertFileStatus2(const struct stat_* src, struct FileStatus2* dst)
+{
+    dst->Dev = (int64_t)src->st_dev;
+    dst->Ino = (int64_t)src->st_ino;
+    dst->Flags = FILESTATUS_FLAGS_NONE;
+    dst->Mode = (int32_t)src->st_mode;
+    dst->Uid = src->st_uid;
+    dst->Gid = src->st_gid;
+    dst->Size = src->st_size;
+
+/* On macOS (https://developer.apple.com/legacy/library/documentation/Darwin/Reference/ManPages/man2/stat.2.html)
+   and BSD (https://www.freebsd.org/cgi/man.cgi?query=stat&sektion=2):
+
+    timespec st_atimespec, st_mtimespec, st_ctimespec, st_birthtimespec
+
+  On Linux (http://man7.org/linux/man-pages/man2/stat.2.html, http://man7.org/linux/man-pages/man2/statx.2.html):
+
+    timespec st_atim, st_mtim, st_ctim
+
+        NOTE: Linux offers birthtime via statx(): we don't currently use it.
+
+  On all Unixes above the time_t st_atime,... st_birthtime are defined to always map to tv_sec part of the timespecs.
+*/
+#ifdef HAVE_STAT_LINUX_FORMAT
```

Comment:
`HAVE_XXX` names should not contain the OS names if possible.

A good name for this would be `HAVE_STAT_TIMESPEC`, like what it is in CoreCLR PAL:

https://github.com/dotnet/coreclr/search?utf8=%E2%9C%93&q=HAVE_STAT_TIMESPEC&type=

Reply:
Ha, seems that the PAL already had all this figured out.